cumulusci/tasks/salesforce/namespaces.py

```diff
@@ -0,0 +1,61 @@
+from cumulusci.core.utils import process_bool_arg
+from cumulusci.utils import inject_namespace
+from typing import Optional, Tuple
+
+namespace_injection_options = {
+    "managed": {
+        "description": "If False, changes namespace_inject to replace tokens with a blank string",
+        "required": False,
+    },
+    "namespaced_org": {
+        "description": "If True, the tokens %%%NAMESPACED_ORG%%% and ___NAMESPACED_ORG___ will get replaced with the namespace.  The default is false causing those tokens to get stripped and replaced with an empty string.  Set this if deploying to a namespaced scratch org or packaging org.",
+        "required": False,
+    },
+    "namespace_inject": {
+        "description": "If set, the namespace tokens in files and filenames are replaced with the namespace's prefix",
+        "required": False,
+    },
+}
+
+
+class NamespaceInjectionMixin:
+    def _inject_namespace(
+        self,
+        filename: str,
+        content: str,
+        namespace: Optional[str] = None,
+        managed: Optional[bool] = None,
+        namespaced_org: Optional[bool] = None,
+    ) -> Tuple[Optional[str], Optional[str]]:
+        _namespace = (
+            namespace
+            or self.options.get("namespace_inject")
+            or self.project_config.project__package__namespace
+        )
+
+        if managed is not None:
+            _managed = managed
+        elif "managed" in self.options:
```

Comment:
I'd suggest also looking for `"unmanaged"`, as many tasks use that option.

(Such tasks wouldn't be able to use this mixin's task options dict by reference, but I think newer tasks should be using the clearer `"managed"` anyway).